aws/resource_aws_kinesis_analyticsv2_application.go

```diff
@@ -0,0 +1,2177 @@
+		Create: resourceAwsKinesisAnalyticsV2ApplicationCreate,
+		Read:   resourceAwsKinesisAnalyticsV2ApplicationRead,
+		Update: resourceAwsKinesisAnalyticsV2ApplicationUpdate,
+		Delete: resourceAwsKinesisAnalyticsV2ApplicationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				arns := strings.Split(d.Id(), ":")
+				name := strings.Replace(arns[len(arns)-1], "application/", "", 1)
+				d.Set("name", name)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_execution_role": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"runtime": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateKinesisAnalyticsV2Runtime,
+			},
+
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"s3_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code_content_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"create_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"last_update_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"cloudwatch_logging_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"log_stream_arn": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+
+			// Flink only
+			"property_groups": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"property_group_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"property_map": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"snapshots_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					snapshotsEnabled := strconv.FormatBool(d.Get(k).(bool))
+					return snapshotsEnabled == old
+				},
+			},
+			"flink_application_configuration": {
```

Comment:
Add an extra level of `application_configuration` configuration block here to match the AWS API.